src/utils.cxx

```diff
@@ -551,6 +564,18 @@ bool hasPathCharacters(const std::string& name) {
     return false;
 }
 
+bool SpackInstalledLib(const std::string& lib) {
+    const std::string prefix = GetSpackEnv("SPACK_INSTALL_PREFIX");
+    if (prefix.empty()) {
+        debug(
+            "Unable to determine Spack install prefix, SPACK_INSTALL_PREFIX "
+            "unset");
+        return false;
+    }
+    std::string stripped_lib = strip_padding(lib);
+    startswith(stripped_lib, prefix);
```

Comment:
Should this normalize (e.g. to make sure that `...\x\.\y` compares equal to `...\x\y`)?